editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-component-renderer.tsx

```diff
@@ -130,6 +131,16 @@ export function createComponentRendererComponent(params: {
       ...appliedProps,
     }
 
+    let spiedVariablesInScope: VariableData = {}
+    if (utopiaJsxComponent.param != null) {
+      for (const paramName of propertiesExposedByParam(utopiaJsxComponent.param)) {
+        spiedVariablesInScope[paramName] = {
+          spiedValue: scope[paramName],
+          insertionCeiling: instancePath,
+        }
+      }
+    }
```

Comment:
This is a bit picky but can this be restructured a bit so that we're not using mutation here to build the `spiedVariablesInScope` above.